model/jpa/src/main/java/org/jboss/aerogear/unifiedpush/jpa/dao/impl/JPABaseDao.java

```diff
@@ -44,6 +47,11 @@ public void setEntityManager(EntityManager entityManager) {
         return entityManager.createQuery(jpql, type);
     }
 
+    protected Query createHibernateQuery(String hql) {
```

Comment:
JPA does not allow an use of scrolling (specifying fetch size / streaming): `org.hibernate.Query.scroll(ScrollMode)`

Forward-only streaming read-only mode is an efficient way to split device tokens into batches.